CONTRIBUTING.md

```diff
@@ -383,6 +383,87 @@ tested it becomes an officially supported Akka feature.
 
 [List of Akka features marked as may change](http://doc.akka.io/docs/akka/current/common/may-change.html)
 
+## Java APIs in Akka
+
+Akka, aims to keep 100% feature parity between the various language DSLs. Implementing even the API for Java in 
+Scala has proven the most viable way to do it, as long as you keep the following in mind:
+
+1. Keep entry points separated in `javadsl` and `scaladsl` unless changing existing APIs which for historical 
+   and binary compatibility reasons does not have this subdivision.
+   
+1. Have methods in the `javadsl` package delegate to the methods in the Scala API. The Akka Stream Scala instances 
+   for example have a `.asJava` method to convert to the `akka.stream.javadsl` counterparts.
+   
+1. When using Scala `object` instances, offer a `getInstance()` method and add a sealed abstract class 
+   (to support Scala 2.11) to get the return type.
+   
+1. When the Scala API contains an `apply` method, use `create` for Java users.
+
+1. Do not nest Scala `object`s more than two levels (as access from Java becomes weird)
+
+1. Be careful to convert values within data structures (eg. for `scala.Long` vs. `java.lang.Long`, use `scala.Long.box(value)`)
+
+1. When compiling with both Scala 2.11 and 2.12, some methods considered overloads in 2.11, become ambiguous in 
+   2.12 as both may be functional interfaces.
+
+1. Complement any methods with Scala collections with a Java collection version
+
+1. Use the `akka.japi.Pair` class to return tuples
+
+1. If the underlying Scala code requires an `ExecutionContext`, make the Java API take an `Executor` and use 
+   `ExecutionContext.fromExecutor(executor)` for conversion.
+
+1. Make use of `scala-java8-compat` conversions, see [GitHub](https://github.com/scala/scala-java8-compat) 
```

Comment:
Add a note that the version of `scala-java8-compat` is locked for bin-compat reasons.
Add a new item to show `akka.util.JavaDurationConverters`.